frappe/utils/data.py

```diff
@@ -1534,7 +1534,10 @@ def money_in_words(
 	number_format = get_number_format()
 
 	fraction_units = frappe.db.get_value("Currency", main_currency, "fraction_units", cache=True)
-	fraction_length = math.ceil(math.log10(fraction_units)) or number_format.precision
+	if fraction_units == 0:
+		fraction_length = number_format.precision
```

Comment:
Can you test what happens when fraction unit is 0? 

E.g. if there's no concept of fraction in VND then we should not just make it up and fallback to number precision with `Cent` as fallback. That sounds stupid and wrong.